block-chain/src/main/java/itx/blockchain/api/DataReader.java

```diff
@@ -0,0 +1,25 @@
+package itx.blockchain.api;
+
+import java.io.IOException;
+
+/**
+ * Data reader used to read data object instance from byte array.
+ * @param <T>
+ */
+public interface DataReader<T> {
+
+    /**
+     * Class type for which this data reader is created.
+     * @return
+     */
+    Class<T> forClass();
+
+    /**
+     * Create an instance of Data class from byte array.
+     * @param data raw serialized instance of data class.
+     * @return instance of data class.
+     * @throws IOException thrown in case reading from byte array goes wrong.
+     */
+    T readData(byte[] data) throws IOException;
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider input validation for data parameter</b></div>
<div id="fix">

Consider adding validation for `null` or empty `data` parameter in the `readData` method interface specification.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
    /**
     * Create an instance of Data class from byte array.
     * @param data raw serialized instance of data class.
     * @return instance of data class.
     * @throws IOException thrown in case reading from byte array goes wrong.
     * @throws IllegalArgumentException thrown if data is null or empty
     */
    T readData(byte[] data) throws IOException;
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/4#issuecomment-2557450846>#0dafab</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged